src/options/options.js

```diff
@@ -97,22 +97,78 @@ document.addEventListener("alpine:init", () => {
       );
       return result;
     },
+    /**
+     * Break array of repositories into chunks of 100, and
+     * store them independently in Chrome storage
+     *
+     * @param {String} workspaceSlug the slug of the Orbit workspace, used to generate a key for the repositories
+     * @param {Array<String>} repositories the un-chunked array of repository names
+     *
+     * @returns Array<String> the addresses of the chunked repositories
+     */
+    _persistRepositories(workspaceSlug, repositories) {
+      // Addresses of the chunked repositories so we can retrieve them later
+      let repository_keys = [];
+      let counter = 1;
+
+      // If there are 240 repositories for workspace "sally", they will be saved as:
+      //
+      // sally:repositories:1 // Repositories 0-100
+      // sally:repositories:2 // Repositories 101-200
+      // sally:repositories:3 // Repositories 201-240
+      while (repositories.length > 0) {
+        let key = `${workspaceSlug}:repositories:${counter}`;
+
+        repository_keys.push(key);
+        chrome.storage.sync.set({
```

Comment:
I had an error running this branch locally, where after saving, running `await chrome.storage.sync.get('repository_keys')` in the Chrome console returned an empty object—indicating those were’t properly saved.

I think it’s coming from the fact that `chrome.storage.sync.set` is asynchronous, and should be called with the `async` keyword (that keyword cascades through the callers):

```diff
diff --git a/src/options/options.js b/src/options/options.js
index 797caf1..5a349ac 100644
--- a/src/options/options.js
+++ b/src/options/options.js
@@ -106,7 +106,7 @@ document.addEventListener("alpine:init", () => {
      *
      * @returns Array<String> the addresses of the chunked repositories
      */
-    _persistRepositories(workspaceSlug, repositories) {
+    async _persistRepositories(workspaceSlug, repositories) {
       // Addresses of the chunked repositories so we can retrieve them later
       let repository_keys = [];
       let counter = 1;
@@ -120,7 +120,7 @@ document.addEventListener("alpine:init", () => {
         let key = `${workspaceSlug}:repositories:${counter}`;

         repository_keys.push(key);
-        chrome.storage.sync.set({
+        await chrome.storage.sync.set({
           [key]: repositories.splice(0, 100),
         });
         counter++;
@@ -140,25 +140,25 @@ document.addEventListener("alpine:init", () => {
      *
      * @returns {token, workspace, repository_keys}
      */
-    _buildStorageObject(token, workspaceSlug, repositories) {
+    async _buildStorageObject(token, workspaceSlug, repositories) {
       const storageObject = {};

       // Set common variables
       storageObject["token"] = token;
       storageObject["workspace"] = workspaceSlug;
-      storageObject["repository_keys"] = this._persistRepositories(
+      storageObject["repository_keys"] = await this._persistRepositories(
         workspaceSlug,
         repositories
       );

       return storageObject;
     },
-    save() {
+    async save() {
       let that = this;
       let repositoriesFullNameForWorkspace =
         this._findAllReposFullNameByWorkspaceSlug();

-      let storageData = this._buildStorageObject(
+      let storageData = await this._buildStorageObject(
         this.token,
         this.selectedWorkspaceSlug,
         repositoriesFullNameForWorkspace
```

The change above fixes the issue on my machine—could you give it a try on your end? Thanks!

Reply:
(Note: the last async might not be necessary, I think Alpine handles async methods transparently)

Reply:
Aaaack right. That might be because my storage was already set from a previous run. Did not think of that. Will manually wipe it between tests

Reply:
I don't think this fix is it. For posterity, I've done everything listed here but also updated the button to have

```
<button id="save" @click="await save()" class="mr-2">
```

instead of

```
<button id="save" @click="save" class="mr-2">
```

But this is not setting the object correctly

![CleanShot 2023-04-21 at 11 42 19@2x](https://user-images.githubusercontent.com/45462299/233616721-e68d63cb-83a2-4923-99a3-31f5848ccd69.png)

See here it's setting the repositories, but none of the other keys that we need, ie `workspace`, `token`, `repsoitory_keys`.

Could you double-check that this code sets the full data object on your end? I can't actually recreate this bug even clearing my storage between runs :(

Reply:
It looks like I had something flaky in my set up—I made sure to be on this branch with no changes, that the storage was empty etc… and it worked like a charm!

However, we might have a backward compatibility issue: folks who set up the integration before we roll this out will see it break (as `chrome.storage.sync.get("repository_keys")` will be empty). My suggestion would be to change `_fetchRepositories` to:

1. Look at the `repository_keys` key—if something’s in there, then we use it
2. If it’s empty, then fallback to the current key

Reply:
🙏🏻 🕺🏻 Excellent shout, did not think of that. Will correct it now ^^